.github/workflows/build.yml

```diff
@@ -0,0 +1,71 @@
+---
+name: Build RPM packages
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_packages:
+    strategy:
+      matrix:
+        build_type:
+          - auto
+          - manual
+        version:
+          - 8
+          - 7
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+          architecture: "x64"
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build packages
+        run: make ${{ matrix.build_type }}${{ matrix.version }}
+
+  publish_packages:
+    needs: build_packages
```

Comment:
Will it make all RPM packages built during parallel `build_packages` jobs available for these parallel `publish_packages` jobs?

I guess we can check it by getting output of `ls -lh _dist8` and `ls -lh _dist7`.

Reply:
Good point, by default they will not share built artifacts. There are two options:
1) consolidate all tasks into single job
2) use GitHub cache action to share data between jobs, and I think this is a better solution. Here's an example how to implement that: https://github.com/stolenng/github-actions-sharing-data/blob/main/.github/workflows/pull_request.yml

Reply:
If we go with the option 1 then we consolidate version 8 packages and version 7 packages separately.

For the option 2 we should check that the cache won't get overridden by parallel jobs. We might have to use several caches to avoid that.

Reply:
I've just tested workflows with cache action and were able to achieve it like this:
```
jobs:
  test_build_job:
    runs-on: ubuntu-20.04
    strategy:
      matrix:
        build_type:
          - auto
          - manual
        version:
          - 8
          - 7
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - run: mkdir _dist${{ matrix.version }}

      - run: touch _dist${{ matrix.version }}/${{ matrix.build_type }}${{ matrix.version }}

      - uses: actions/cache@v2
        id: restore-build
        with:
          path: _dist${{ matrix.version }}/*
          key: ${{ github.sha }}-${{ matrix.build_type }}${{ matrix.version }}

  test_publish_job:
    needs: test_build_job
    runs-on: ubuntu-20.04
    strategy:
      max-parralel: 1   # not sure if necessary, depends on how happy package_cloud would be with 4 simultaneous uploads
      matrix:
        build_type:
          - auto
          - manual
        version:
          - 8
          - 7
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - uses: actions/cache@v2
        id: restore-build
        with:
          path: _dist${{ matrix.version }}/*
          key: ${{ github.sha }}-${{ matrix.build_type }}${{ matrix.version }}

      - run: ls -la _dist${{ matrix.version }}
```
The drawback here is that 4 separate publish jobs are created, but it didn't work properly when same cache was simultaneously used in build jobs.

Reply:
Adjusting publish job following way allows to achieve the same by single job:
```
  test_publish_job:
    needs: test_build_job
    runs-on: ubuntu-20.04
    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - uses: actions/cache@v2
        with:
          path: _dist7/*
          key: ${{ github.sha }}-auto7

      - uses: actions/cache@v2
        with:
          path: _dist7/*
          key: ${{ github.sha }}-manual7

      - uses: actions/cache@v2
        with:
          path: _dist8/*
          key: ${{ github.sha }}-auto8

      - uses: actions/cache@v2
        with:
          path: _dist8/*
          key: ${{ github.sha }}-manual8

      - run: ls -la _dist7 _dist8
```
```
> Run ls -la _dist7 _dist8
_dist7:
total 8
drwxr-xr-x  2 runner docker 4096 Feb 15 16:22 .
drwxr-xr-x 48 runner docker 4096 Feb 15 16:22 ..
-rw-r--r--  1 runner docker    0 Feb 15 16:22 auto7
-rw-r--r--  1 runner docker    0 Feb 15 16:22 manual7

_dist8:
total 8
drwxr-xr-x  2 runner docker 4096 Feb 15 16:22 .
drwxr-xr-x 48 runner docker 4096 Feb 15 16:22 ..
-rw-r--r--  1 runner docker    0 Feb 15 16:22 auto8
-rw-r--r--  1 runner docker    0 Feb 15 16:22 manual8
```

Reply:
Nice! Thank you for figuring it out.